From c02f3d042e7952974e5936514426b2a4571057f2 Mon Sep 17 00:00:00 2001
From: Lyude Paul <lyude@redhat.com>
Date: Sun, 4 Nov 2018 20:32:48 -0500
Subject: [PATCH 097/151] meson.build: Set dep_x11 and dep_xext for panfrost

Using cc.find_library() for non-C stdlib dependencies only works if
you're guaranteed to have all of the dependencies you need installed
system wide. This isn't the case for cross compilation, which means that
we need to do things the right way so that the correct flags for linking
get specified during cross compilation. So, do that.

This, with the addition of

https://github.com/mesonbuild/meson/pull/4473

Fix cross compilation of panfrost for my aarch64 system
---
 meson.build                              | 3 +++
 src/gallium/drivers/panfrost/meson.build | 7 ++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 9c09dff..d392e60 100644
--- a/meson.build
+++ b/meson.build
@@ -1362,6 +1362,9 @@ if with_platform_x11
     dep_xfixes = dependency('xfixes')
     dep_xcb_glx = dependency('xcb-glx', version : '>= 1.8.1')
     dep_xxf86vm = dependency('xxf86vm')
+  elif with_gallium_panfrost == 'panfrost'
+    dep_x11 = dependency('x11')
+    dep_xext = dependency('xext')
   endif
   if (with_any_vk or with_glx == 'dri' or
        (with_gallium_vdpau or with_gallium_xvmc or with_gallium_va or
diff --git a/src/gallium/drivers/panfrost/meson.build b/src/gallium/drivers/panfrost/meson.build
index 9a55b6d..08cc6d8 100644
--- a/src/gallium/drivers/panfrost/meson.build
+++ b/src/gallium/drivers/panfrost/meson.build
@@ -56,7 +56,12 @@ midgard_nir_algebraic_c = custom_target(
 libpanfrost = static_library(
   'panfrost',
   [files_panfrost, midgard_nir_algebraic_c],
-  dependencies: [cc.find_library('X11', required: true), cc.find_library('Xext', required: true), dep_thread, idep_nir],
+  dependencies: [
+    dep_x11,
+    dep_xext,
+    dep_thread,
+    idep_nir
+  ],
   include_directories : [inc_common, inc_gallium_aux, inc_gallium, inc_include, inc_src, include_directories('midgard'), include_directories('include')],
   c_args : [c_vis_args, c_msvc_compat_args],
 )
-- 
2.7.4

